code/game/machinery/big_manipulator.dm

```diff
@@ -234,6 +388,64 @@
 /obj/machinery/big_manipulator/proc/finish_rotate_animation(backward)
 	animate(manipulator_hand, transform = matrix(180 * backward, MATRIX_ROTATE), working_speed*0.5)
 
+/obj/machinery/big_manipulator/proc/check_filter(obj/item/what_item)
+	var/filtered_obj = filter_obj?.resolve()
+	if(!istype(what_item, selected_type))
+		return
+	/// We use filter only on items. closets, humans and etc don't need filter check.
+	if(istype(what_item, /obj/item))
+		if((filtered_obj && !istype(what_item, filtered_obj)))
+			return FALSE
+	return TRUE
+
+/// Create dummy to force him use our item and then delete him.
+/obj/machinery/big_manipulator/proc/create_abstract_living()
+	var/mob/living/carbon/human/abstract_living = new /mob/living/carbon/human(get_turf(src))
+	abstract_living.icon_state = "blank"
+	abstract_living.alpha = 0
+	abstract_living.name = src.name
+	return abstract_living
+
+/// Proc called when we changing item interaction mode.
+/obj/machinery/big_manipulator/proc/change_mode()
+	manipulate_mode++
+	if(manipulate_mode > THROW_ITEM_MODE)
+		manipulate_mode = DROP_ITEM_MODE
+	update_priority_list()
+	is_work_check()
+
+/// Update priority list in ui. Creating new list and sort it by priority number.
+/obj/machinery/big_manipulator/proc/update_priority_list()
+	allowed_priority_settings = list()
+	var/list/priority_mode_list = list()
+	if(manipulate_mode == DROP_ITEM_MODE)
+		priority_mode_list = priority_settings_for_drop.Copy()
+	if(manipulate_mode == USE_ITEM_MODE)
+		priority_mode_list = priority_settings_for_use.Copy()
+	if(isnull(priority_mode_list))
+		return
+	for(var/we_need_increasing in 1 to priority_mode_list.len)
+		for(var/datum/manipulator_priority/what_priority in priority_mode_list)
```

Comment:
```suggestion
		for(var/datum/manipulator_priority/what_priority as always in priority_mode_list)
```
The lists are only populated by instances of this datum, right?

Reply:
erm there is no as always mey be you mean as anything